ang/contactlayout/contactlayout.js

```diff
@@ -507,4 +552,48 @@
     };
   });
 
+  // Service for loading relationship type options and displaying loading state.
+  angular.module('contactlayout')
+    .service('contactLayoutRelationshipOptions', function (crmApi) {
+      var relationshipOptionsPromise;
+      var service = this;
+
+      service.isLoading = false;
+      service.options = [];
+
+      // loads and stores the relationship type options.
+      service.loadOptions = function () {
+        service.isLoading = true;
+
+        if (!relationshipOptionsPromise) {
+          relationshipOptionsPromise = crmApi('RelationshipType', 'get', {
+            is_active: 1,
+            sequential: 1,
+            options: { limit: 0 }
+          });
+        }
+
+        return relationshipOptionsPromise
+          .then(formatRelationshipOptions)
+          .then(function (options) {
+            service.isLoading = false;
+            service.options = options;
+          })
+      };
+
+      // for each relationship type, it includes an option for the a_b relationship
+      // and another for the b_a relationship.
+      function formatRelationshipOptions (relationshipTypeResponse) {
+        return _.chain(relationshipTypeResponse.values)
```

Comment:
Note that when switching api to api4, it returns the values directly you don't have to drill down into response.values